radix-engine-stores/src/hash_tree/mod.rs

```diff
@@ -23,10 +22,29 @@ mod test;
 #[allow(dead_code)]
 mod types;
 
+/// Part of the ID addressing the upper layer of the JMT.
+pub type DbIndex = Vec<u8>;
+
+/// Part of the ID addressing the lower layer of the JMT.
+pub type DbKey = Vec<u8>;
+
+/// A complete ID of the substate, as tracked by the JMT.
+pub struct DbId {
```

Comment:
Relatedly, the interface in `interface.rs` exposes an ordering but allows the user to specify a Mapper?

I don't think this interface is correct - the mapper has to be fixed to expose a deterministic lexicographic ordering. So I'm happy if the JmtMapper actually becomes DbKeyMapper / a more general database mapper as part of commonizing the logic, and we get rid of the `DatabaseMapper` interface.

Reply:
I like the rename suggestions (although a bit long) and will apply them 👍

I also considered the wrapper types (instead of typedefs), but then the code changes propagate a bit further (because yeah, no 0-syntax conversion). It still makes total sense, but then I'd really prefer to promote these types up (i.e. to be used by the other DB impls - I guess this is your other suggestion too). Let's first get the builders team's 🟢 for that, though?

> So I'm happy if the JmtMapper actually becomes DbKeyMapper / a more general database mapper as part of commonizing the logic, and we get rid of the DatabaseMapper interface.

It's probably a natural thing to do when we promote the key types mentioned above to be used by all DBs - I will automatically do it too if I do the above 👍

Reply:
as agreed on Slack:

Let's merge this PR as-is and wait for the related https://github.com/radixdlt/radixdlt-scrypto/pull/978 to merge, and then we will discuss and do the deep renames.